nestedtensor/csrc/functions.cpp

```diff
@@ -219,27 +218,56 @@ NestedTensor interpolate(NestedTensor input,
         throw error;
     }
 
+    TensorNode input_structure = input.get_structure();
     auto options = F::InterpolateFuncOptions().mode(int_mode);
-    if (size.has_value()) {
-      options = options.size(size.value().vec());
-    }
-    
-    if (scale_factor.has_value()) {
-      options = options.scale_factor(scale_factor.value().vec());
-    }
-
     if (align_corners.has_value()) {
       options.align_corners() = align_corners.value();
     }
 
-    TensorNode res = map(
+    // Either scale factor or size can be passed
+    if (scale_factor.has_value()) {
+      options = options.scale_factor(scale_factor.value().vec());
+      TensorNode res = map(
         [&options](at::Tensor input_tensor) {
           return F::interpolate(input_tensor.unsqueeze(0), options).squeeze(0);
         },
         input_structure);
+      return NestedTensor(std::move(res));
+    }
 
-    return NestedTensor(std::move(res));
-}
+    // Get input leaves count
+    auto fn = [](at::Tensor leaf, int64_t input) {
+      return input + 1;
+    };
+    auto leaves_count = reduce<decltype(fn), int64_t, at::Tensor>(input.get_structure(), fn, 0);
+    
+    if (size.has_value()) {
+      if (size.value().size() != 1 && size.value().size() != leaves_count) {
+        std::stringstream error;
+        error << "Interpolate has to take either 1 size tuple or same amount as leaves in Nested Tensor." << std::endl;
+        throw error;
+      }
+
+      int size_i = 0;
+      TensorNode res = map(
+          [&options, &size_i, &size](at::Tensor input_tensor) {
+            if(size.value().size() == 1) {
+              options = options.size(size.value()[0]);
+            } else {
+              options = options.size(size.value()[size_i]);
+              size_i++;
```

Comment:
This will break in the future if we change the way map iterate over leaves (for example for performance reasons) as far as I can tell. I assume you're trying to associate one entry of size.value() to each Tensor, if it's not of length one and otherwise want to broadcast the value across all Tensors.

If this is restricted to input NestedTensor's of nested dimension 1 and in the case where there is one size each per Tensor, you could create a ```NestedNode<std::vector<int64_t>>``` via ```size_node = NestedNode<std::vector<int64_t>>(size.value())```. You can then pass size_node as a third argument to map and change the signature of the lambda you're expanding from

```
[&options, &size_i, &size](at::Tensor input_tensor)
```

to

```
[&options](at::Tensor input_tensor, std::vector<int64_t> size)
```

Now, for the other case you could just pass ```size.value()[0]``` to the lambda closure by creating a new value (maybe ```size_one_value = size.value()[0]```) and changing your lambda signature to 

```
[&options, &size_one_value](at::Tensor input_tensor) 
```


Of course, overall, this means writing two separate map statements and branching on whether ```size.value().size() == 1``` or not.